### PR TITLE
Np 1263 rewrite scenarios

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: src/test/resources/features/Roles.feature [hunk 1]
```diff
@@ -2,33 +2,23 @@ Feature: Role features
...
-    And the request contains a JSON body with following key-value pairs
-      | key      | value   |
-      | rolename | theRole |
-      | type     | Role    |
-    When the authorized client sends the request to add a new Role
-    Then a new role is stored in the database
-    And the description of the role is returned to the authorized client
+    When the AuthorizedClient requests to add the NewRole to the Database
+    Then an OK message is returned
+    And a role description is returned
```
Comment (janbj) [inline src/test/resources/features/Roles.feature]: If you want to describe the role description you can do that in one of the scenarios

File: src/test/resources/features/Users.feature [hunk 2]
```diff
@@ -1,119 +1,84 @@
...
+
+    And a non-existing NewUser with username "newUser" that does not yet exist in the database
+    And NewUser is specified that it will have the roles:
+      | roles |
+      | RoleA |
+
+  Scenario: AuthorizedClient adds a new user
+    When the AuthorizedClient sends a request to add the NewUser to the Database
+    Then the NewUser is added to the database
+    And the response object contains the UserDescription of the NewUser
```
Comment (janbj) [inline src/test/resources/features/Users.feature]: UserDescription here and "role description" in other file

File: src/test/resources/features/Users.feature [hunk 3]
```diff
@@ -1,119 +1,84 @@
...
   Scenario: Authorized client gets an existing user
-    Given  that a user entry with the username "someone@institution" exists in the database
-    And the authorized client forms a "GET" request
-    And the request has the following path parameters:
-      | parameter | value               |
-      | username  | someone@institution |
-    When the authorized client sends the request to read the user
-    Then a user description is returned
+    When the AuthorizedClient sends a request to read the ExistingUser from the Database
+    Then the response object contains the UserDescription of the ExistingUser
```
Comment (janbj) [inline src/test/resources/features/Users.feature]: in the response the UserDescription is the object

File: src/test/resources/features/Users.feature [hunk 4]
```diff
@@ -1,119 +1,84 @@
...
-      | role-name |
-      | RoleB     |
-    When the authorized client sends the request to update the user
+    When the AuthorizedClient requests to update the ExistingUser and sets the following roles:
+      | roles |
+      | roleB |
+      | roleC |
+    Then the ExistingUser is updated asynchronously
+    And a Location header with the ExistingUser URI is included in the response
+    And the ExistingUser contains a list of roles with the following role-names:
```
Comment (janbj) [inline src/test/resources/features/Users.feature]: Tidy up roles and role-names

File: src/test/resources/features/Users.feature [hunk 5]
```diff
@@ -1,119 +1,84 @@
...
-    And the request body also contains a list of roles with the following role-names
-      | role-name |
-      | RoleB     |
-    When the authorized client sends the request to update the user
+    When the AuthorizedClient requests to update the ExistingUser and sets the following roles:
+      | roles |
+      | roleB |
+      | roleC |
+    Then the ExistingUser is updated asynchronously
+    And a Location header with the ExistingUser URI is included in the response
```
Comment (janbj) [inline src/test/resources/features/Users.feature]: And the response has a Location header with ExistingUser's URI as value

File: src/test/resources/features/Users.feature [hunk 6]
```diff
@@ -1,119 +1,84 @@
...
-      | key         | value               |
-      | username    | someone@institution |
-      | institution | institution         |
-      | type        | User                |
-    And the request body also contains a list of roles with the following role-names
-      | role-name |
-      | RoleB     |
-    When the authorized client sends the request to update the user
+    When the AuthorizedClient requests to update the ExistingUser and sets the following roles:
+      | roles |
```
Comment (janbj) [inline src/test/resources/features/Users.feature]: And the request body has the following roles

**Review by janbj — DISMISSED** [review]
It's hard to say what's most correct for scenarios. I have provided some suggestions based on how I like it, but it's your final decision.

File: src/test/resources/features/Users.feature (hunk 6 above)
Comment (axthosarouris) [reply src/test/resources/features/Users.feature]: This would create a misunderdansting before the API expects a whole User object and not only the roles.

**Review by janbj — APPROVED** [review]

**Pull request merged**